run_project_tests.py

```diff
@@ -1230,19 +1236,20 @@ def _run_tests(all_tests: T.List[T.Tuple[str, T.List[TestDef], bool]],
     # Ensure we only cancel once
     tests_canceled = False
 
-    # Optionally enable the tqdm progress bar
+    # Optionally enable the tqdm progress bar, but only if there is at least one TestRunFuture
     global safe_print
     futures_iter: T.Iterable[RunFutureUnion] = futures
-    try:
-        from tqdm import tqdm
-        futures_iter = tqdm(futures, desc='Running tests', unit='test')
+    if len(futures) > 2:
```

Comment:
Comment and code don’t match

Reply:
futures is a RunFutureUnion, which should always have one `LogRunFuture` (either `Not running {name} tests.` or `Running {name} tests.`) and one or more `TestRunFuture`s -- I only care about the latter.

Arguably I could check how many of them are isinstance TestRunFuture, but since I "know" that LogRunFuture is only used once for one header per category, I took a shortcut and hardcoded that there should be more than 2 of the union of types.

Would you prefer switching from `len(futures)` to checking how many of them are specifically TestRunFuture?

Reply:
Or perhaps that assumption could at least be commented better.

Reply:
That would be good, otherwise looks good to me

Reply:
Fixed.